src/controllers/message.js

```diff
@@ -40,17 +40,45 @@ const getAllMessages = async (req, res) => {
 
         if (otherUser) {
             if (otherUser in acc) {
-                //If other user already exist in messages array, it will push the new message to it.
-                acc[otherUser].push(curr)
+                // If other user already exist in messages array, it will push the new message to it.
+                acc[otherUser].messages.push(curr)
             } else {
-                //otherwise it will create new array of messages!
-                acc[otherUser] = [curr]
+                // otherwise it will create new object
+                acc[otherUser] = {
+                    userId: otherUser,
+                    username: curr.otherUserName,
+                    messages: [curr],
+                }
```

Comment:
I like how you set this up. username is seems to be undefined though. Just pull username by userId here and I think we can remove the code added below.

Reply:
@jpremo I tried to pull username by userId , but it return username: undefined on my terminal since I log the acc. I assume it's because we define otherUser as userId string. I tried to convert otherUser as object and use async in reduce method to find the Id by using await in User model, but it returns an empty object even though the status is 200. Any idea on this?

 ```
acc[otherUser] = {
                    userId: otherUser,
                    username: userId.username,
                    messages: [curr],
                }
```
<img width="283" alt="image" src="https://user-images.githubusercontent.com/108501073/236016041-e79bdbd3-17c8-4b76-a5db-e6a5db95228e.png">